node_api/src/services/entity.service.ts

```diff
@@ -301,4 +301,37 @@ export default class EntityService {
       })
     ).data;
   }
+
+  async findEntitiesBySnomedCodes(codes: string[]) {
+    const iris = codes.map(code => SNOMED.NAMESPACE + code);
+    const result = await this.getPartialEntities(iris, [RDFS.LABEL, IM.CODE]);
+    return result.map(resolved => resolved.data);
+  }
+
+  async findValidatedEntitiesBySnomedCodes(codes: string[]) {
+    const entities = await this.findEntitiesBySnomedCodes(codes);
+    const needed = await this.getDistillation(entities as TTIriRef[]);
```

Comment:
entities is of type string[]. Fix type if you want it as TTIriRef[], don't just cast